java/compaction-status-store/src/test/java/sleeper/compaction/status/task/QueryAllCompactionTasksIT.java

```diff
@@ -0,0 +1,61 @@
+/*
+ * Copyright 2022 Crown Copyright
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package sleeper.compaction.status.task;
+
+import org.junit.Test;
+import sleeper.compaction.status.testutils.DynamoDBCompactionTaskStatusStoreTestBase;
+import sleeper.compaction.task.CompactionTaskStatus;
+
+import java.time.Instant;
+
+import static org.assertj.core.api.Assertions.assertThat;
+
+public class QueryAllCompactionTasksIT extends DynamoDBCompactionTaskStatusStoreTestBase {
+
+    @Test
+    public void shouldReportMultipleCompactionTasksSortedMostRecentFirst() {
+        // Given
+        CompactionTaskStatus task1 = taskWithStartTime(
+                Instant.parse("2022-10-06T11:18:00.001Z"));
+        CompactionTaskStatus task2 = taskWithStartAndFinishTime(
+                Instant.parse("2022-10-06T11:19:00.001Z"),
+                Instant.parse("2022-10-06T11:21:00.001Z"));
+        CompactionTaskStatus task3 = taskWithStartTime(
+                Instant.parse("2022-10-06T11:20:00.001Z"));
+        CompactionTaskStatus task4 = taskWithStartAndFinishTime(
+                Instant.parse("2022-10-06T11:21:00.001Z"),
+                Instant.parse("2022-10-06T11:22:00.001Z"));
+
+        // When
+        store.taskStarted(task1);
+        store.taskStarted(task2);
+        store.taskFinished(task2);
+        store.taskStarted(task3);
```

Comment:
With the dates defined above, it makes sense for task3 to start before task2 finishes, which would swap the positions of task3 and task2 in the assertion

Reply:
Thanks, I've changed that. It doesn't change the assertion because it's ordering by the start date.